packages/core/src/utils/ReturnableObjectPool.ts

```diff
@@ -0,0 +1,28 @@
+import { IPoolElement, ObjectPool } from "./ObjectPool";
+
+/**
+ * Returnable Object Pool.
+ */
+export class ReturnableObjectPool<T extends IPoolElement> extends ObjectPool<T> {
+  private _lastElementIndex: number;
```

Comment:
According to your design, `ReturnableObjectPool` does not allow calling `garbageCollection`